src/TorchSharp/Hub.cs

```diff
@@ -21,17 +22,46 @@ public static partial class hub
             /// <exception cref="InvalidDataException">SHA256 hash doesn't match</exception>
             public static void download_url_to_file(string url, string dst, string hash_prefix = null, bool progress = true)
             {
-                using (var webClient = new WebClient()) {
-                    if (progress) {
-                        // TODO: Implement progress.
-                        // Console.WriteLine("download_url_to_file(progress: true) is not supported.");
-                    }
-                    webClient.DownloadFile(url, dst);
-                    string fileHash = GetFileChecksum(dst);
-                    if (hash_prefix != null && !fileHash.StartsWith(hash_prefix)) {
-                        throw new InvalidDataException($"invalid hash value (expected \"{hash_prefix}\", got \"{fileHash}\"");
+                try {
+                    Task.Run(async () => {
+                        await download_url_to_file_async(url, dst, hash_prefix, progress);
```

Comment:
@NiklasGustafsson 

I'm not very familiar to this, but I think this is needed to avoid deadlock in case the method is called from ASP.NET or desktop apps.

https://docs.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development#the-thread-pool-hack

Probably ASP.NET Core doesn't have this issue.

https://blog.stephencleary.com/2017/03/aspnetcore-synchronization-context.html#:~:text=ASP.NET%20Core%20does%20not%20have%20a%20SynchronizationContext%2C%20so,thread%2C%20and%20they%20may%20all%20run%20in%20parallel.